### DISCUSSION TITLE
GitHub in my company : can we use it ?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#42119] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello,

I am a PhD engineer in chemical and process engineering, specialized in modeling, simulation and optimization of industrial processes. I develop computational tools in VBA, FORTRAN 90, C, Python and Matlab for my company, with 5 other colleagues. We are not developers in the way of software developers, and we don't know the classical workflow used in development teams like Git/GitHub. We write equations, collect physical and chemical data, and compute it all in calculation tools that help us in our work.

Today we realize that we need a way to manage our code, their version, their validation etc... So I've been looking into using Git/GitHub for our team.

Git seems to be already ready to be used among us, it is not very user friendly (some of us don't want to work with the console to check the change and validate the new version).

GitHub, on the other hand, is very user-friendly and could help us in many ways. There is only one problem we are concerned about: our company absolutely refuses the idea of our code being stored in a cloud that is not ours (privacy and security issues).

Here's my question: is there a solution to use GitHub, but with our code in our cloud/server?

Thanks in advance for your time, and sorry for any mistakes in my message, English is only my second language.

**Comment by samharp** [comment]
My old workplace had a similar conundrum where they were using an outdated version-control system and were looking to using GitHub, but couldn't get over the fact that Microsoft (owner of GitHub) would hold all of their potentially-sensitive data (even if it was stored in private repos). To give you their answer to the problem, they decided to go with self-hosted GitLab. Personally, I find GitHub much more user-friendly, but GitLab does have a UI, you don't have to manage everything from console, plus you can set up your own instance to manage code versioning and other Git features.
